MantidQt/CustomInterfaces/test/ReflDataProcessorPresenterTest.h

```diff
@@ -0,0 +1,633 @@
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460_monitors"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_monitors"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TRANS_13463"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TRANS_13464"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TRANS_13463_13464"));
+
+    // Tidy up
+    AnalysisDataService::Instance().remove("TestWorkspace");
+    AnalysisDataService::Instance().remove("IvsLam_13460_0_10");
+    AnalysisDataService::Instance().remove("IvsLam_13460_10_20");
+    AnalysisDataService::Instance().remove("IvsLam_13460_20_30");
+    AnalysisDataService::Instance().remove("IvsLam_13462_0_10");
+    AnalysisDataService::Instance().remove("IvsLam_13462_10_20");
+    AnalysisDataService::Instance().remove("IvsLam_13462_20_30");
+    AnalysisDataService::Instance().remove("IvsQ_13460_0_10");
+    AnalysisDataService::Instance().remove("IvsQ_13460_10_20");
+    AnalysisDataService::Instance().remove("IvsQ_13460_20_30");
+    AnalysisDataService::Instance().remove("IvsQ_13462_0_10");
+    AnalysisDataService::Instance().remove("IvsQ_13462_10_20");
+    AnalysisDataService::Instance().remove("IvsQ_13462_20_30");
+    AnalysisDataService::Instance().remove("IvsQ_13460_0_10_13462_0_10");
+    AnalysisDataService::Instance().remove("IvsQ_13460_10_20_13462_10_20");
+    AnalysisDataService::Instance().remove("IvsQ_13460_20_30_13462_20_30");
+    AnalysisDataService::Instance().remove("IvsQ_binned_13460_0_10");
+    AnalysisDataService::Instance().remove("IvsQ_binned_13460_10_20");
+    AnalysisDataService::Instance().remove("IvsQ_binned_13460_20_30");
+    AnalysisDataService::Instance().remove("IvsQ_binned_13462_0_10");
+    AnalysisDataService::Instance().remove("IvsQ_binned_13462_10_20");
+    AnalysisDataService::Instance().remove("IvsQ_binned_13462_20_30");
+    AnalysisDataService::Instance().remove("TOF_13460");
+    AnalysisDataService::Instance().remove("TOF_13462");
+    AnalysisDataService::Instance().remove("TOF_13460_0_10");
+    AnalysisDataService::Instance().remove("TOF_13460_10_20");
+    AnalysisDataService::Instance().remove("TOF_13460_20_30");
+    AnalysisDataService::Instance().remove("TOF_13462_0_10");
+    AnalysisDataService::Instance().remove("TOF_13462_10_20");
+    AnalysisDataService::Instance().remove("TOF_13462_20_30");
+    AnalysisDataService::Instance().remove("TOF_13460_monitors");
+    AnalysisDataService::Instance().remove("TOF_13462_monitors");
+    AnalysisDataService::Instance().remove("TRANS_13463");
+    AnalysisDataService::Instance().remove("TRANS_13464");
+    AnalysisDataService::Instance().remove("TRANS_13463_13464");
```

Comment:
You may consider a single command like
`AnalysisDataService::Instance().clear();` instead of a bunch of `AnalysisDataService::Instance().remove("...");` commands. Be careful, there is no notification and nothing happens when calling remove and for so many names typos are likely.
Clear lives like doesExist and retrieve in `DataService.h`. I checked that doesExist fails after calling clear. Remove will remove the name of the workspace. Clear will empty the DataService.